extension/src/panel.ts

```diff
@@ -102,25 +99,34 @@ export class HistoryDebuggerPanel {
     // Set the webview's initial html content
     this.update()
 
-    // Listen for when the panel is disposed
-    // This happens when the user closes the panel or when the panel is closed programatically
-    this.panel.onDidDispose(() => this.dispose(), null, this.disposables)
-
+    let reloadServer: http.Server | undefined = undefined
     // Start a local HTTP server to automatically reload the webview when rollup completes.
     if (process.env.TEMPORAL_DEBUGGER_EXTENSION_DEV_MODE) {
-      const server = http.createServer((_req, res) => {
+      reloadServer = http.createServer((_req, res) => {
         void vscode.commands.executeCommand("workbench.action.webview.reloadWebviewAction")
         res.writeHead(200, "OK")
         res.end()
       })
-      server.listen(55666, "127.0.0.1")
+      reloadServer.listen(55666, "127.0.0.1")
     }
+
+    // Listen for when the panel is disposed
+    // This happens when the user closes the panel or when the panel is closed programatically
+    this.panel.onDidDispose(async () => {
+      // Close local servers
+      server.terminate()
+      if (reloadServer) {
+        reloadServer.close()
+      }
+
+      // Clear timeouts
+      this.timeouts.forEach((timeout) => clearTimeout(timeout))
+
+      await this.dispose(), null, this.disposables
+    })
   }
 
   public async dispose(): Promise<void> {
```

Comment:
`onDidDipose` is a vscode event listener that will trigger the callback (which includes `this.dispose` when the panel is closed, so we actually want both. 
We currently don't have anything in `this.disposables`* so we could just remove that, but I opted to keep it in case we add any in the future.

*The only functions that currently return a `Disposable` are in the `activate` function and are added to `subscriptions` which are 
> an array to which disposables can be added. When this extension is deactivated the disposables will be disposed.

[source](https://code.visualstudio.com/api/references/vscode-api)